server/app/uk/gov/ons/addressIndex/server/controllers/AddressController.scala

```diff
@@ -1,106 +1,131 @@
+import uk.gov.ons.addressIndex.model.server.response.Container
+import uk.gov.ons.addressIndex.model.server.response.Model.HybridResponse
 
 @Singleton
 class AddressController @Inject()(
-  override val esRepo: ElasticsearchRepository,
+  override val esRepo: ElasticSearchRepository,
   parser: AddressParserModule,
   conf : AddressIndexConfigModule
 )(implicit override val ec: ExecutionContext) extends AddressIndexController with AddressIndexActions {
 
   val logger = Logger("address-index-server:AddressController")
 
   /**
-    * Test elastic is connected
     *
+    * @param uprn
+    * @param format
     * @return
     */
-  def elasticTest(): Action[AnyContent] = Action async { implicit req =>
-    esRepo.client execute {
-      get cluster health
-    } map { resp =>
-      Ok(resp.toString)
+  def uprnQuery(uprn: String, format: Option[String]): Action[AnyContent] = Action async { implicit req =>
+
+    logger info s"#addressQuery:\nuprn $uprn , format: ${format.getOrElse("no format supplied")}"
+
+    uprnSearch(
+      uprn = uprn,
+      format = format flatMap(_.stringToScheme)
+    ) map { r =>
+      jsonOk(
+        Container.fromHybridResponse(
+          optAddresses = Some(r.as[HybridResponse].toSeq),
+          tokens = Seq.empty,
+          status = PredefStatus.Ok,
+          limit = 1,
+          offset = 0,
+          total = 1,
+          maxScore = r.maxScore
+        )
+      )
     }
   }
 
   /**
-    * Address query API
-    *
-    * @param input  the address query
-    * @param format requested format of the query (paf/nag)
-    * @return Json response with addresses information
+    * @param input
+    * @param format
+    * @param offset
+    * @param limit
+    * @return
     */
-  def addressQuery(input: String, format: String, offset: Option[String] = None, limit: Option[String] = None): Action[AnyContent] = Action async { implicit req =>
-    logger info s"#addressQuery:\ninput $input , format: $format , offset: ${offset.getOrElse("default")}, limit: ${limit.getOrElse("default")}"
-   // get the defaults and maxima for the paging parameters from the config
+  def addressQuery(
+    input: String,
+    format: Option[String] = None,
+    offset: Option[String] = None,
+    limit: Option[String] = None
+  ): Action[AnyContent] = Action async { implicit req =>
+
+    logger info s"#addressQuery:\n" +
+      s"input $input , format: $format , offset: ${offset.getOrElse("default")}, limit: ${limit.getOrElse("default")}"
+
+    //get the defaults and maxima for the paging parameters from the config
     val defLimit = conf.config.elasticSearch.defaultLimit
     val defOffset = conf.config.elasticSearch.defaultOffset
     val maxLimit = conf.config.elasticSearch.maximumLimit
     val maxOffset = conf.config.elasticSearch.maximumOffset
-// TODO Look at refactoring to use types
-    val limval = limit.getOrElse(defLimit.toString())
-    val offval = offset.getOrElse(defOffset.toString())
+
+    //TODO REMOVE This
```

Comment:
Should remove the code below and use the scala type system.
specifically;
limit: String  -> limit: Int
offset: String -> offset: Int